### ISSUE TITLE
Specification: what happens on callback failures?

### ISSUE THREAD

**Issue by andre-merzky** [ExaWorks/job-api-spec#133]
The specification is currently silent on implementation behavior when state notification callbacks raise an exception.

I suggest that we define that exceptions are ignored (but `SHOULD` be logged).  It is the responsibility of the application to make sure the cb is safe.

**Comment by hategan** [comment]
Yes, I think an implementation note encouraging implementations to ignore but loudly log CB exceptions is a good idea.
